### PR TITLE
Feature/verify on submit

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Verify on Submit 

verify-on-submit

### CODE REVIEW

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    toolbarGutters: {
+        padding: '0 16px',
+    },
+    containerWrapper: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'start',
+        flex: '1 1 0',
+        backgroundColor: 'white',
+        height: 'calc(100vh - 64px)',
```
Comment (ektasawant) [inline src/pages/form-validation/verify-on-submit/index.tsx:35]: Could you please add `height: 'calc(100vh - 56px)',` for mobile?

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import { EmptyState } from '@pxblue/react-components';
+
+type OnChangeHandler = InputProps['onChange'];
+
+const useStyles = makeStyles((theme: Theme) => ({
+    appbarRoot: {
+        padding: 0,
+    },
+    toolbarGutters: {
+        padding: '0 16px',
```
Comment (ektasawant) [inline src/pages/form-validation/verify-on-submit/index.tsx:27]: can we change it to 

padding: `0 ${theme.spacing(4)}px`,

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+                    <Slide
+                        direction="left"
+                        in={showDeviceAddedScreen}
+                        mountOnEnter
+                        unmountOnExit
+                        timeout={slideAnimationDurationMs}
+                    >
+                        <div
+                            style={{ display: 'flex', alignItems: 'center', height: '100%', justifyContent: 'center' }}
```
Comment (ektasawant) [inline src/pages/form-validation/verify-on-submit/index.tsx:214]: Can we move it to style section?

**Comment by ektasawant** [comment]
Button height is changing if spinner is visible. Can we match height of button?
image

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                    <Slide
+                        direction="right"
+                        in={showAddDeviceScreen}
+                        mountOnEnter
+                        unmountOnExit
+                        timeout={deviceAdded ? slideAnimationDurationMs : 0}
+                    >
+                        <div>
+                            <Typography variant={'h6'} style={{ marginBottom: theme.spacing(2) }}>
+                                Find Device
```
Comment (ektasawant) [inline src/pages/form-validation/verify-on-submit/index.tsx:159]: Can we change button text `Find Device`  => `Search Device` ?

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+                            </Typography>
+                            <Typography variant={'body1'} style={{ marginBottom: theme.spacing(4) }}>
+                                For the sake of this example, serial number 123 will yield a successful device search.
```
Comment (ektasawant) [inline src/pages/form-validation/verify-on-submit/index.tsx:162]: 123 is bold.

File: src/pages/form-validation/verify-on-submit/index.tsx (hunk 2 above)
Comment (emclaug2) [reply src/pages/form-validation/verify-on-submit/index.tsx:27]: Ok I've updated this spacing.

File: src/pages/form-validation/verify-on-submit/index.tsx (hunk 3 above)
Comment (emclaug2) [reply src/pages/form-validation/verify-on-submit/index.tsx:214]: Yep will do.

File: src/pages/form-validation/verify-on-submit/index.tsx (hunk 4 above)
Comment (emclaug2) [reply src/pages/form-validation/verify-on-submit/index.tsx:159]: Ok this is updated.

File: src/pages/form-validation/verify-on-submit/index.tsx (hunk 5 above)
Comment (emclaug2) [reply src/pages/form-validation/verify-on-submit/index.tsx:162]: Now it's bold.

**Comment by emclaug2** [comment]
All items you mentioned @ektasawant have been addressed; ready for another round of review whenever you have time.

**Comment by ektasawant** [comment]
> All items you mentioned @ektasawant have been addressed; ready for another round of review whenever you have time.

Hi @emclaug2 , thanks for addressing comments. Looks good .

@huayunh , could you please let us know in mobile  `Search device` button should take full width?  Since In angular design pattern I followed design https://pxblue.github.io/patterns/forms#verifying-on-action (for mobile) which shows button at right side similar to desktop view.

**Comment by emclaug2** [comment]
Cool! This example is already using the full-width button so there should be any changes required here.

**Review by jeffvg — APPROVED** [review]
looks good to me

**Comment by daileytj** [comment]
Screen Shot 2021-09-07 at 12 31 05 PM

After submitting an invalid code, the input is in an error state with error text that says 'device not found'... this is as expected. However, on change the error text/state persists when it should be cleared.

And... there's some merge conflicts.

**Comment by emclaug2** [comment]
Ok @daileytj the merge conflicts are resolved & the error is now cleared onChange.

**Review by huayunh — APPROVED** [review]
Functions well visually

File: src/pages/form-validation/verify-on-submit/index.tsx [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+                                    }
+                                }}
+                            />
+
+                            <div className={classes.buttonContainer}>
+                                <Button
+                                    className={classes.submitButton}
+                                    color={'primary'}
+                                    variant={'contained'}
+                                    startIcon={<>{!loading && <Search className={classes.searchIcon} />}</>}
```
Comment (huayunh) [inline src/pages/form-validation/verify-on-submit/index.tsx:181]: You shouldn't need this searchIcon class anymore now that you are using the startIcon class (the icon size can be 16, can be 20, not a significant issue here)

Comment (emclaug2) [reply src/pages/form-validation/verify-on-submit/index.tsx]: Ah okay, I've removed the searchIcon class and the default's looking good.

**Pull request merged**